### PR TITLE
feat(v3): remove deprecated checks and calls

### PR DESCRIPTION
**Pull request by JacobCoffee**
## Description

- Initial work of removing the deprecated things in v3

### CODE REVIEW

**Comment by JacobCoffee** [comment]
Think i might need help on the new openapi stuff, i wish i had the docs rendered :P

**Comment by JacobCoffee** [comment]
#3264 should give us the rendered docs

File: litestar/response/base.py [hunk 1]
```diff
@@ -88,11 +87,6 @@ def __init__(
         status_code = status_code or HTTP_200_OK
         self.headers = MutableScopeHeaders()
 
-        if encoded_headers is not None:
```
Comment (provinzkraut) [inline litestar/response/base.py:91]: I think all of these changes ought to remove the deprecated parameter as well, not just the deprecation warning :P

File: litestar/static_files/config.py
Comment (provinzkraut) [inline litestar/static_files/config.py:1]: I think this should be a separate PR because it means we can remove all the special casing for the static files app in our routing logic as well

**Comment by JacobCoffee** [comment]
make it smaller :)
